bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/LatestVersionImplementationPicker.java

```diff
@@ -21,15 +21,14 @@
 import java.util.regex.Pattern;
 
 import org.apache.sling.models.spi.ImplementationPicker;
-import org.osgi.framework.Constants;
 import org.osgi.service.component.annotations.Component;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.osgi.service.component.propertytypes.ServiceRanking;
 
 /**
  * Picks latest model version of a core component model
  */
-@Component(property=Constants.SERVICE_RANKING+":Integer=1") // this must come after ResourceTypeBasedResourcePicker
+@Component
+@ServiceRanking(value = 1)// this must come after ResourceTypeBasedResourcePicker
```

Comment:
This will only work together with Sling Model Impl 1.4.12 and later (https://github.com/apache/sling-org-apache-sling-models-impl/commit/89ae8beba7f6bd03e3af13c45124cbe62f122b3b), in older versions the ResourceTypeBasedResourcePicker was registered at `Integer.MAX_VALUE - 1`. Not sure if all supported AEM versions already come with that version.

Reply:
Since v2.11.0 the core components depend on 1.4.12 of the org.apache.sling.models.impl (see system requirements in README)